src/utilities/embeddable-utils.ts

```diff
@@ -1,38 +1,100 @@
 import { v4 as uuidv4 } from "uuid";
-import { IAuthoringMetadata, IRuntimeMetadata } from "@concord-consortium/lara-interactive-api";
 import {
   IExportableAnswerMetadata, IManagedInteractive, IReportState, IExportableMultipleChoiceAnswerMetadata,
   IExportableOpenResponseAnswerMetadata, IExportableInteractiveAnswerMetadata, IExportableImageQuestionAnswerMetadata,
   Embeddable
 } from "../types";
+import { getPortalData } from "../firebase-db";
 
 export const isQuestion = (embeddable: Embeddable) =>
   (embeddable.type === "ManagedInteractive" && embeddable.library_interactive?.data?.enable_learner_state) ||
   (embeddable.type === "MwInteractive" && embeddable.enable_learner_state);
 
+// LARA uses a map from the answer type to a question type.
+// Instead of using this map, we look directly at the authoredState this ought to give us more
+// flexibility to support more special types in the report without needing to update
+// the activity player
+export const questionType = (rawAuthoredState: string | null | undefined): string => {
+  if (!rawAuthoredState) {
+    return "iframe_interactive";
+  }
+
+  // There is a IAuthoringMetadata type, but because we don't know if this has
+  // valid authored state or not, that type isn't not used here.
+  let authoredState: any = {};
+  try {
+    authoredState = JSON.parse(rawAuthoredState);
+  } catch (e) {
+    // this isn't valid JSON, so we use the default value of {} for the authored state
+  }
+
+  if (!authoredState.questionType) {
+    return "iframe_interactive";
+  }
+
+  return authoredState.questionType as string;
+};
+
+export const remoteEndpoint = (): string => {
+  const portalData = getPortalData();
+  if (!portalData) {
+    return "";
+  }
+
+  // This is a way to verify the portalData is of type IPortalData
+  if (portalData.type === "authenticated") {
+    return portalData.runRemoteEndpoint;
```

Comment:
[Type guards](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards) can be useful for doing run-time type checks and then returning that information into the type system. Could potentially be useful here and/or with the untyped authored state above.